packages/core/src/domain/configuration/experimentalFeatures.spec.ts

```diff
@@ -0,0 +1,39 @@
+import {
+  updateExperimentalFeatures,
+  isExperimentalFeatureEnabled,
+  resetExperimentalFeatures,
+} from './experimentalFeatures'
+
+describe('experimentalFeatures', () => {
+  it('initial state is empty', () => {
+    resetExperimentalFeatures()
+    expect(isExperimentalFeatureEnabled(undefined as any)).toBeFalse()
+    expect(isExperimentalFeatureEnabled('foo')).toBeFalse()
+    expect(isExperimentalFeatureEnabled('bar')).toBeFalse()
+    expect(isExperimentalFeatureEnabled('')).toBeFalse()
+  })
+
+  it('updateExperimentalFeatures', () => {
+    updateExperimentalFeatures(['foo'])
+    expect(isExperimentalFeatureEnabled('foo')).toBeTrue()
+    expect(isExperimentalFeatureEnabled('bar')).toBeFalse()
+  })
+
+  it('resetExperimentalFeatures clears state', () => {
+    resetExperimentalFeatures()
+    expect(isExperimentalFeatureEnabled('foo')).toBeFalse()
+    expect(isExperimentalFeatureEnabled('bar')).toBeFalse()
+  })
```

Comment:
IMO since it is only meant to be used in unit tests, no need to add tests for it